master/internal/api_config_policies.go

```diff
@@ -0,0 +1,118 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/types/known/structpb"
+	"gopkg.in/yaml.v3"
+
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+)
+
+// workload types are used to distinguish configuration for experiments and NTSCs.
+const (
+	// Should not be used.
+	WorkloadTypeUnspecified = "UNKNOWN"
+	// Configuration policies for experiments.
+	WorkloadTypeExperiment = "EXPERIMENT"
+	// Configuration policies for NTSC.
+	WorkloadTypeNTSC = "NTSC"
+)
+
+func validWorkloadEnum(val string) bool {
+	switch val {
+	case WorkloadTypeExperiment, WorkloadTypeNTSC:
+		return true
+	default:
+		return false
+	}
+}
+
+var yamlString = `
+invariant_configs:
+  description: "test"
+constraints:
+  resources:
+    max_slots: 4
+  priority_limit: 10
+ `
+
+func stubData() (*structpb.Struct, error) {
+	// put yaml string into a map
+	var yamlMap map[string]interface{}
+	if err := yaml.Unmarshal([]byte(yamlString), &yamlMap); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal yaml: %v", err)
```

Comment:
Nit: On the off chance that this stub code lives on by being changed and not just deleted, maybe we wanna wrap stuff like this for detection via `errors.Is()`?

```suggestion
		return nil, fmt.Errorf("unable to unmarshal yaml: %w", err)
```